optuna/integration/allennlp.py

```diff
@@ -9,17 +10,142 @@
 import optuna
 from optuna._experimental import experimental
 from optuna._imports import try_import
+from optuna import load_study
+from optuna import Trial
+
 
 with try_import() as _imports:
     import allennlp
     import allennlp.commands
     import allennlp.common.util
-    from allennlp.training import EpochCallback
 
+# EpochCallback is conditionally imported for CI because allennlp is not available on
+# the eivironment that builds documentation.
 if _imports.is_successful():
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+    # `allennlp.training.EpochCallback` is a subclass of `Registerable`
+    # https://docs.allennlp.org/master/api/training/trainer/#epochcallback
+    # On the other hand, `EpochCallback` defined here is not `Registerable`,
+    # which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback.
+
+        This class is introduced for documentation CI.
+
+        """
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+PPID = os.getppid()
```

Comment:
A comment will surely help future maintainers, especially as you're using the parent id and not the id of the current process. Could you explain how this variable and the environment variables are used to communicate between processes?

Reply:
I created `_add_ppid` and added some notes there. 6af736b :pray: